core/src/types/ibc.rs

```diff
@@ -84,6 +102,53 @@ mod ibc_rs_conversion {
             })
         }
     }
+
+    impl From<IbcShieldedTransfer> for Memo {
+        fn from(shielded: IbcShieldedTransfer) -> Self {
+            let bytes =
+                shielded.try_to_vec().expect("Encoding shouldn't failed");
+            HEXLOWER.encode(&bytes).into()
+        }
+    }
+
+    impl TryFrom<Memo> for IbcShieldedTransfer {
+        type Error = Error;
+
+        fn try_from(memo: Memo) -> Result<Self> {
+            let bytes = HEXLOWER
+                .decode(memo.as_ref().as_bytes())
+                .map_err(Error::DecodingHex)?;
+            Self::try_from_slice(&bytes)
+                .map_err(Error::DecodingShieldedTransfer)
+        }
+    }
+
+    /// Get the shielded transfer from the memo
+    pub fn get_shielded_transfer(
+        event: &IbcEvent,
+    ) -> Result<Option<IbcShieldedTransfer>> {
+        if event.event_type != "fungible_token_packet" {
```

Comment:
is the string `"fungible_token_packet"` part of any ICS (maybe ICS 20)? either way, it would be best to assign a constant binding to it, IMO